### PR TITLE
add uah with castling

### PR DESCRIPTION
**Pull request by EllAchE**

### CODE REVIEW

File: cjsmin/src/chess.ts [hunk 1]
```diff
@@ -733,7 +766,10 @@ export class Chess {
             from: square,
             to: this._epSquare,
             piece: PAWN,
-            captured: PAWN,
+            capture: {
```
Comment (bennyrubanov) [inline cjsmin/src/chess.ts:769]: and this? why capture vs captured

File: cjsmin/src/chess.ts [hunk 2]
```diff
@@ -811,7 +850,7 @@ export class Chess {
     this._board[sq] = {
       type: type as PieceType,
       color: color as Color,
-      unambiguousSymbol: 'pa',
+      unambiguousSymbol,
```
Comment (bennyrubanov) [inline cjsmin/src/chess.ts:853]: not sure i understand this change either. why was it set to 'pa' initially, and why remove this?

File: cjsmin/src/chess.ts [hunk 3]
```diff
@@ -1458,19 +1514,30 @@ export class Chess {
         } else {
           index = move.to + 16;
         }
-        throw new Error('TODO: Implement en passant capture');
+        // determine captured pawn based on square of captured piece
+        const square = algebraic(move.to);
+        let unambiguousSymbol = '';
+        if (square.charAt(1) === '2') {
+          unambiguousSymbol += 'P';
```
Comment (bennyrubanov) [inline cjsmin/src/chess.ts:1521]: good catch on setting white pawns to be capitalized and black to be lowercased, thx

**Review by bennyrubanov — APPROVED** [review]
approve review, but not sure i understand all the changes atm

File: cjsmin/src/chess.ts (hunk 2 above)
Comment (EllAchE) [reply cjsmin/src/chess.ts:853]: this is the hardcoding I mentioned. This was done so we could compile while requiring this and left as a TODO For this PR

File: cjsmin/src/chess.ts (hunk 1 above)
Comment (EllAchE) [reply cjsmin/src/chess.ts:769]: originally I kept both capure & captured for backwards compatability. Didn't update the name again after I changed that

File: cjsmin/src/chess.ts [hunk 4]
```diff
@@ -571,17 +604,17 @@ function strippedSan(move: string) {
...
+  _turn: Color = WHITE;
+  _header: Record<string, string> = {};
+  _kings: Record<Color, number> = { w: EMPTY, b: EMPTY };
+  _epSquare = -1;
+  _halfMoves = 0;
+  _moveNumber = 0;
+  _history: History[] = [];
+  _castling: Record<Color, number> = { w: 0, b: 0 };
+
+  constructor() {
```
[the remark this answers is not in the document]
Comment (EllAchE) [reply cjsmin/src/chess.ts:616]: make the properties publicly accessible so we can debug

File: cjsmin/src/chess.ts (hunk 3 above)
Comment (EllAchE) [reply cjsmin/src/chess.ts:1521]: I actually think this isn't necessary, reverting it to just use capture.UAS

**Pull request merged**